packages/core-plugin-api/src/extensions/extensions.tsx

```diff
@@ -152,7 +152,14 @@ export function createReactExtension<
   data?: Record<string, unknown>;
   name?: string;
 }): Extension<T> {
-  const { data = {} } = options;
+  const { data = {}, name } = options;
+  if (!name) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      'Declaring extensions without name is DEPRECATED. ' +
```

Comment:
There's a potential to move this down and add componentName in the message. However it was just "Component" in the cases I saw when testing so I refrained from using it

Reply:
Hmmm. It would be really nice to get some form of useful info. I wis we'd be able to use like `__filename` or something instead of `'Component'` below, is that or some other smart approach possible @Rugvip ?

Reply:
not really, it's why we need this in the first place 😅, could be that a stack trace might help out tho?

Reply:
There's some help from the stacktrace from `console.warn` can we do better?